site/content/docs/commands/deploy.en.md

```diff
@@ -5,33 +5,57 @@ $ copilot deploy
 
 ## What does it do?
 
-This command is used to run either [`copilot svc deploy`](../commands/svc-deploy.en.md) or [`copilot job deploy`](../commands/job-deploy.en.md) under the hood. The steps involved in `copilot deploy` are the same as those involved in `copilot svc deploy` and `copilot job deploy`:
+This command is used to run either [`copilot svc deploy`](../commands/svc-deploy.en.md) or [`copilot job deploy`](../commands/job-deploy.en.md) under the hood.
 
-1. When `image.build` exists in the manifest:
+It also checks for service and environment manifests in the local workspace, and allows you to initialize and deploy workloads and environments.
+
+If a workload is uninitialized, `--init-wkld` will initialize the workload before deploying it.
+
+If the desired environment is uninitialized, you may initialize it with `--init-env`. 
+
+The `--deploy-env` flag can be specified to skip environment deployment confirmation, or can be set to false (`--deploy-env=false`) to skip 
+deploying the environment.
+
+The steps involved in `copilot deploy` are the same as those involved in `copilot svc deploy` and `copilot job deploy`:
+
+1. If your service does not exist, optionally initialize it (via `--init-env[=false]`).
+2. If the target environment does not exist, optionally initialize it with custom credentials (via `--init-env[=false]`, `--region`, `--profile`, or `--aws-*` flags).
+3. Optionally deploy the environment before service deployment (via `--deploy-env[=false]`).
+4. When `image.build` exists in the manifest:
     1. Build your local Dockerfile into an image
     2. Tag it with the value from `--tag` or the latest git sha (if you're in a git directory)
     3. Push the image to ECR
-2. Package your manifest file and addons into CloudFormation
-3. Create / update your ECS task definition and job or service.
+5. Package your manifest file and addons into CloudFormation.
+6. Create / update your ECS task definition and job or service.
 
 ## What are the flags?
 
 ```
       --allow-downgrade                Optional. Allow using an older version of Copilot to update Copilot components
                                        updated by a newer version of Copilot.
   -a, --app string                     Name of the application.
-      --detach                         Optional. Skip displaying CloudFormation deployment progress.
+      --aws-access-key-id string       Optional. An AWS access key for the environment account.
+      --aws-secret-access-key string   Optional. An AWS secret access key for the environment account.
+      --aws-session-token string       Optional. An AWS session token for temporary credentials.
+      --deploy-env bool                Deploy the target environment before deploying the workload.
+      --detach bool                    Optional. Skip displaying CloudFormation deployment progress.
```

Comment:
Do we need these `bool` at the end?

Reply:
I think so, just for consistency with other flag types. I'd prefer to keep it explicit even if the help docs don't render them when you do `copilot deploy -h`.

Reply:
I feel like the `bool` at the end might be misleading for `copilot deploy --detach true`...but just a nit im ok for both!